shard.go

```diff
@@ -180,14 +179,31 @@ func marshalValues(values map[uint8]interface{}) []byte {
 		}
 
 		// Encode value after field id.
-		// TODO: Support non-float types.
 		switch v := v.(type) {
 		case float64:
+			buf = make([]byte, 9)
 			binary.BigEndian.PutUint64(buf[1:9], math.Float64bits(v))
+		case bool:
+			buf = make([]byte, 2)
```

Comment:
@benbjohnson -- from implementing this function, it seems to me we need to code more information in a (field ID, value) pair -- we must also encode the type. Can you confirm? `unmarshalValues` just assumes that it's a float type in the database, but that will no longer be true with these changes.

If I am correct, a single byte after the field ID would be sufficient.